path.go

```diff
@@ -87,7 +87,14 @@ func (p *Path) RedirectRoot() error {
 // It will use custom regex to parse the path if it's provided in
 // the given record.
 func zoneFromPath(r *http.Request, rec record) (string, int, []string, error) {
-	path := fmt.Sprintf("%s?%s", r.URL.Path, r.URL.RawQuery)
+	path := r.URL.Path
+
+	// Check if request is from a Git client
+	if strings.HasPrefix(r.Header.Get("User-Agent"), "git") {
+		path = "/" + strings.Split(r.URL.Path, "/")[1]
```

Comment:
What is your reasoning behind this?

Reply:
Git sends the request to a path like `/torproxy/info/refs?service=git-upload-pack` and if we don't split the first part of the path in here it would lead to this:
```
2019-09-30T22:19:24.772+03:30 [INFO] 127.0.0.1:57746 - 62342 "TXT IN _redirect.git.example.test. udp 44 false 512" NOERROR qr,aa,rd 207 0.000153389s
2019-09-30T22:19:24.773+03:30 [INFO] 127.0.0.1:35984 - 37431 "TXT IN _redirect.refs.info.torproxy.git.example.test. udp 63 false 512" NXDOMAIN qr,aa,rd 144 0.000183976s
2019-09-30T22:19:24.773+03:30 [INFO] 127.0.0.1:34249 - 24113 "TXT IN _redirect._.info.torproxy.git.example.test. udp 60 false 512" NXDOMAIN qr,aa,rd 141 0.000245351s
2019-09-30T22:19:24.774+03:30 [INFO] 127.0.0.1:46892 - 57040 "TXT IN _redirect._._.torproxy.git.example.test. udp 57 false 512" NXDOMAIN qr,aa,rd 138 0.000276778s
2019-09-30T22:19:24.774+03:30 [INFO] 127.0.0.1:49182 - 12356 "TXT IN _redirect._._._.git.example.test. udp 50 false 512" NXDOMAIN qr,aa,rd 131 0.000249735s
```

Reply:
Just thought of a better way to handle it in case the current code causes some edge cases.
We can get the index of `/info/refs` and split the path from start to that index.